pkg/vault/test/foundry/BatchRouterERC4626Pool.t.sol

```diff
@@ -0,0 +1,409 @@
+
+        dai.mint(bob, erc4626PoolInitialAmount);
+        dai.approve(address(waDAI), erc4626PoolInitialAmount);
+        waDAI.deposit(erc4626PoolInitialAmount, bob);
+
+        usdc.mint(bob, erc4626PoolInitialAmount);
+        usdc.approve(address(waUSDC), erc4626PoolInitialAmount);
+        waUSDC.deposit(erc4626PoolInitialAmount, bob);
+
+        _initPool(
+            erc4626Pool,
+            [erc4626PoolInitialAmount, erc4626PoolInitialAmount].toMemoryArray(),
+            erc4626PoolInitialBPTAmount - MIN_BPT
+        );
+
+        IERC20(address(erc4626Pool)).approve(address(permit2), MAX_UINT256);
+        permit2.approve(address(erc4626Pool), address(router), type(uint160).max, type(uint48).max);
+        permit2.approve(address(erc4626Pool), address(batchRouter), type(uint160).max, type(uint48).max);
+
+        IERC20(address(erc4626Pool)).approve(address(router), type(uint256).max);
+        IERC20(address(erc4626Pool)).approve(address(batchRouter), type(uint256).max);
+
+        vm.stopPrank();
+    }
+
+    modifier checkBuffersWhenStaticCall(address sender) {
+        uint256 beforeUSDCBalance = usdc.balanceOf(sender);
+        uint256 beforeDAIBalance = dai.balanceOf(sender);
+        (uint256 beforeWaUSDCBufferBalanceUnderlying, uint256 beforeWaUSDCBufferBalanceWrapped) = vault
+            .getBufferBalance(waUSDC);
+        (uint256 beforeWaDAIBufferBalanceUnderlying, uint256 beforeWaDAIBufferBalanceWrapped) = vault.getBufferBalance(
+            waDAI
+        );
+
+        _;
+
+        uint256 afterUSDCBalance = usdc.balanceOf(sender);
+        assertEq(beforeUSDCBalance - afterUSDCBalance, 0, "USDC balance should be the same");
+
+        uint256 afterDAIBalance = dai.balanceOf(sender);
+        assertEq(beforeDAIBalance - afterDAIBalance, 0, "DAI balance should be the same");
+
+        (uint256 afterWaUSDCBufferBalanceUnderlying, uint256 afterWaUSDCBufferBalanceWrapped) = vault.getBufferBalance(
+            waUSDC
+        );
+        assertEq(
+            beforeWaUSDCBufferBalanceWrapped - afterWaUSDCBufferBalanceWrapped,
+            0,
+            "waUSDC wrapped buffer balance should be the same"
+        );
+        assertEq(
+            afterWaUSDCBufferBalanceUnderlying - beforeWaUSDCBufferBalanceUnderlying,
+            0,
+            "waUSDC underlying buffer balance should be the same"
+        );
+
+        (uint256 afterWaDAIBufferBalanceUnderlying, uint256 afterWaDAIBufferBalanceWrapped) = vault.getBufferBalance(
+            waDAI
+        );
+        assertEq(
+            beforeWaDAIBufferBalanceWrapped - afterWaDAIBufferBalanceWrapped,
+            0,
+            "waDAI wrapped buffer balance should be the same"
+        );
+        assertEq(
+            afterWaDAIBufferBalanceUnderlying - beforeWaDAIBufferBalanceUnderlying,
+            0,
+            "waDAI underlying buffer balance should be the same"
+        );
+    }
+
+    function testAddLiquidityUnbalancedToERC4626Pool() public {
```

Comment:
We're checking the buffers and the LP's BPT balance, but we should also check the pool balance for the boosted pool (amount of wrapped tokens that leave buffers should be added to the boosted pool).